src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -3290,6 +3298,10 @@ public ILocalSymbol GetDeclaredSymbol(CatchDeclarationSyntax catchDeclaration, C
                             symbols = ImmutableArray.Create<Symbol>(ReducedExtensionMethodSymbol.Create(symbol));
                             resultKind = conversion.ResultKind;
                         }
+                        else if (conversion.ConversionKind == ConversionKind.NoConversion && conversion.Operand is BoundUnconvertedAddressOfOperator unconvertedAddressOf)
+                        {
+                            symbols = GetMethodGroupSemanticSymbols(unconvertedAddressOf.Operand, boundNodeForSyntacticParent, binderOpt, out resultKind, out isDynamic, out memberGroup);
```

Comment:
>symbols = GetMethodGroupSemanticSymbols(unconvertedAddressOf.Operand, boundNodeForSyntacticParent, binderOpt, out resultKind, out isDynamic, out memberGroup); [](start = 28, length = 158)

It is not obvious what scenario we are trying to address here. Could you please elaborate?

Reply:
If there wasn't a conversion, the default behavior is get the `ExpressionSymbol`, if there is one. For unconverted address of operators, this isn't correct: we want to get all the symbols from the method group that underlies the address of. I can add this as a comment in source.

---
In reply to: [428854021](https://github.com/dotnet/roslyn/pull/44436#discussion_r428854021) [](ancestors = 428854021)

Reply:
Examples might help. Could you point to specific unit-test(s)?

---
In reply to: [428885643](https://github.com/dotnet/roslyn/pull/44436#discussion_r428885643) [](ancestors = 428885643,428854021)

Reply:
`PublicApi_GetSymbolInfo`, where the conversion is incorrect. Because the conversion fails, we don't end up creating a method group conversion and we have a `BoundUnconvertedAddressOfOperator` left. I'll add an additional tests to further exercise this path around attempting to get the address of a missing method.

---
In reply to: [428891527](https://github.com/dotnet/roslyn/pull/44436#discussion_r428891527) [](ancestors = 428891527,428885643,428854021)

Reply:
I could also be persuaded that the underlying identifier syntax should be the thing that returns the method symbols, and not the whole `&M` syntax. I'm not aware of any particular api precedence we have in this area, so your opinion would be appreciated.

---
In reply to: [428896674](https://github.com/dotnet/roslyn/pull/44436#discussion_r428896674) [](ancestors = 428896674,428891527,428885643,428854021)

Reply:
> I could also be persuaded that the underlying identifier syntax should be the thing that returns the method symbols, and not the whole &M syntax. I'm not aware of any particular api precedence we have in this area, so your opinion would be appreciated.

I think it is reasonable to return a symbol for ```&M``` syntax. However, we should still decide what, if anything, should be returned for the ```M```. We also have an API called GetMemberGroup (or something similar). I think it should be applicable to ```M``` syntax. Consider checking what we do right now for method group to delegate conversion, it might be good to be consistent in some way even though we have an extra syntax node involved. A regular invocation is another scenario to check.

---
In reply to: [429357637](https://github.com/dotnet/roslyn/pull/44436#discussion_r429357637) [](ancestors = 429357637,428896674,428891527,428885643,428854021)

Reply:
This is now handled correctly. My existing solution worked for `GetSymbolInfo`, but ran into issues with `GetMemberGroup`. I've updated it to handle `BoundUnconvertedAddressOfOperator` directly, which removes the need for this special handling here. The only adjustments we need to make are around successful method-group-to-function-pointer conversions.

---
In reply to: [429412739](https://github.com/dotnet/roslyn/pull/44436#discussion_r429412739) [](ancestors = 429412739,429357637,428896674,428891527,428885643,428854021)